docs/new_flux_ideas_and_guide.md

```diff
@@ -0,0 +1,985 @@
+
+### Increase
+Increase works on columns that are counters. Like the rate function, it is called by itself (not as an argument to 
+aggregate or selector). Its arguments `by` and `window` operate in the same way that rate does. It will calculate 
+the total increase accounting for resets in the counter (note this increase is not like the old Flux one). For 
+example, this could be used to determine the total number of bytes send over a network card in a given window of time.
+
+```
+increase = (table, columns, by = [], window = 0) = {}
+```
+
+### Filter
+Filter will filter a result set to the requested columns and records matching the predicate criteria.
+
+```
+// function signature
+filter = (table, fn) => {}
+
+// calling it
+data |> filter(fn: (r) => r.foo == "bar") // will return all columns in data, but only rows where the value of foo is bar
+
+data |> filter({fn: (r) => r.foo == "bar" and r.time > -3h and r.time < -1h) // will return only rows in specified time range
+```
+
+### Drop
+Drop will drop the specified columns from the table, including system columns if requested:
+```
+// function signature
+drop = (table, columns) => {}
+
+// calling it
+data |> drop(["foo", "_key"])
+```
+
+### Keep
+Keep will keep only the specified columns in the table, while keeping any system columns (starting with `_`):
+
+```
+// function signature
+keep = (table, columns) => {}
+
+// calling it
+data |> keep(["foo", "time"])
+```
+
+### Rename
+Rename will rename a column from one thing to another. Note that the `_key` values will not be altered even 
+if that column appears in the key. Here's the function signature and how it can be called:
+
+```
+// function signature
+rename = (table, from, to) => {}
+
+// calling it
+data |> rename(from: "foo", to: "bar")
+```
+
+### Graphing
+In order for the visualization front end to know which rows group into a series, you can call the `graph` function 
+that will include in the result set information about what constitutes a series that should be graphed. This function 
+must be called if you will be visualizing with any kind of graph with an x and y axis. Nothing need be done for 
+tables and single stat panes. In the future, Flux will have functions instructing the front end to visualize and 
+structure the data in any of the ways it supports, like single stat panes, tables, bar charts, line graphs, heatmaps, 
+histograms and many more.
+
+Here's an example of the basic graph call:
+
+```
+@my_data.cpu{host, usage_system, usage_user, time, cpu == "cpu-total" AND time > -1h}
+    |> graph(yCols: ["usage_system", "usage_user"], xCol: "time", by: ["host"])
```

Comment:
What if we make the UI collect this information from the user? Meaning Flux returns a table and the UI has to either figure out based on context or ask the user what the `x` and `y` columns are and if there is any grouping? This way the decision is still explicit but doesn't require that we design an graphing API inside of Flux.